### ISSUE TITLE
scatter drops points if xscale is activated TOO LATE

### ISSUE THREAD

**Issue by untom** [matplotlib/matplotlib#3059]
I'm using matplotlib 1.3.1 with GTKagg backend. I found that if I call `ax.set_xscale` _before_ calling `ax.scatter` everything works fine. However if I change the order of the commands, most points are not displayed. Here is some sample code:

```
fig, ax = plt.subplots(1, 2)
n = 1000
x = np.abs(np.random.laplace(size=n))
y = np.arange(n)
ax[0].scatter(x, y)
ax[0].set_xscale('log')
ax[1].set_xscale('log')
ax[1].scatter(x, y)
```

This leads to the following output:  http://imgur.com/xqsR9fr

I'm not sure if this is the same as #2872 or not, maybe someone who has the current master installed can test that.

**Comment by tacaswell** [comment]
On current master and with v3.3.0 this produces 

![image](https://user-images.githubusercontent.com/199813/89944279-8991fe80-dbed-11ea-8b03-ebdf42ef21b5.png)

Close (and not chasing back to when we actually fixed this...)

**Comment by QuLogic** [comment]
It's fixed in 3.2.2 at least, so I'd wager it's the lazy autoscaling work in 3.2.0.
